MAVProxy/modules/mavproxy_chat/chat_openai.py

```diff
@@ -87,71 +96,74 @@ def set_api_key(self, api_key_str):
 
     # send text to assistant
     def send_to_assistant(self, text):
-        # check connection
-        if not self.check_connection():
-            return "chat: failed to connect to OpenAI"
-
-        # create a new message
-        input_message = self.client.beta.threads.messages.create(
-            thread_id=self.assistant_thread.id,
-            role="user",
-            content=text
-        )
-        if input_message is None:
-            return "chat: failed to create input message"
-
-        # create a run
-        self.run = self.client.beta.threads.runs.create(
-            thread_id=self.assistant_thread.id,
-            assistant_id=self.assistant.id
-        )
-        if self.run is None:
-            return "chat: failed to create run"
-
-        # wait for run to complete
-        run_done = False
-        while not run_done:
-            # wait for one second
-            time.sleep(0.1)
+        # get lock
+        with self.send_lock:
```

Comment:
You could queue instead of lock - all callers have to handle the async nature of the connection with OpenAI anyway.

Reply:
ok interesting, I'll look into that.